File: Builder/Golang/PDFReportBuilder.go
package main

type PDFReportBuilder struct {
	report *Report
}

func (pdfReportBuilder *PDFReportBuilder) buildContent() {
	if pdfReportBuilder.report == nil {
		pdfReportBuilder.report = &Report{}
	}

	pdfReportBuilder.report.setContent("Summary")
}

func (pdfReportBuilder *PDFReportBuilder) buildFile() {
	if pdfReportBuilder.report == nil {
		pdfReportBuilder.report = &Report{}
	}

	pdfReportBuilder.report.setFilename("Report.pdf")
}

func (pdfReportBuilder *PDFReportBuilder) getResult() *Report {
	return pdfReportBuilder.report
}

File: Builder/Dart/PDFReportBuilder.dart
import 'IReportBuilder.dart';
import 'Report.dart';

class PDFReportBuilder implements IReportBuilder {
  Report? _report;

  @override
  void buildContent() {
    if (_report == null) {
      _report = Report();
    }

    _report?.setContent("Summary");
  }

  @override
  void buildFile() {
    if (_report == null) {
      _report = Report();
    }

    _report?.setFilename("Report.pdf");
  }

  @override
  Report? getResult() {
    return _report;
  }
}
